modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/30_search.yml

```diff
@@ -482,3 +482,79 @@
                         }]
     - match: { error.root_cause.0.type: "illegal_argument_exception" }
     - match: { error.root_cause.0.reason: "script score function must not produce negative scores, but got: [-9.0]"}
+
+---
+"Do not throw exception if input score is negative":
+  - do:
+      index:
+        index: test
+        id: 1
+        body: { "color" : "orange red yellow" }
+  - do:
+      index:
+        index: test
+        id: 2
+        body: { "color": "orange red purple", "shape": "red square" }
+  - do:
+      index:
+        index: test
+        id: 3
+        body: { "color" : "orange red yellow purple" }
+  - do:
+      indices.refresh: { }
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            function_score:
+              query:
+                multi_match:
+                  query: "red"
+                  type: "cross_fields"
+                  fields: [ "color", "shape^100"]
+                  tie_breaker: 0.1
+              functions: [{
+                "script_score": {
+                  "script": {
+                    "lang": "painless",
+                    "source": "_score"
+                  }
+                }
+              }]
+          explain: true
+  - match: { hits.total.value: 3 }
+  - match: { hits.hits.2._score: 0.0 }
```

Comment:
nice, so the functional score on a document will be at least zero, instead of negative. This will fix the score per field level. 

Wondering if this also fix the per field level score is negative? can we run profile api on yaml test and check perFieldSimilarity score is negative or zero?

Reply:
The logic focuses on the input score passed to a function or script, regardless of where that score came from.

Both before and after this change, we have checks that the score **returned** by a function or script is not negative (and throw an exception if it is). Unfortunately, this meant that a function/script that just passes a score through would throw an exception if the input score is negative.

So, for **any** case where a function or script gets an illegal negative value as input, this change defensively treats it as zero.

Reply:
awesome! thank you